From b12b8a2b93dc0232d4af98a084f32fb716c19d26 Mon Sep 17 00:00:00 2001
From: Michal Schmidt <mschmidt@redhat.com>
Date: Mon, 29 Oct 2012 21:59:34 +0100
Subject: [PATCH] units: agetty overrides TERM

Environment=TERM=... has no effect on agetty who sets it by itself. To
really set TERM to a specified value, it has to be given on the command
line.

https://bugzilla.redhat.com/show_bug.cgi?id=870622
(cherry picked from commit 2161de72c517d34d1ceb9b4c1a300f0b54ce5a9c)
---
 units/getty@.service.m4        | 3 +--
 units/serial-getty@.service.m4 | 3 +--
 2 files changed, 2 insertions(+), 4 deletions(-)

diff --git a/units/getty@.service.m4 b/units/getty@.service.m4
index fcf1de1..9ee3bf5 100644
--- a/units/getty@.service.m4
+++ b/units/getty@.service.m4
@@ -40,9 +40,8 @@ IgnoreOnIsolate=yes
 ConditionPathExists=/dev/tty0
 
 [Service]
-Environment=TERM=linux
 # the VT is cleared by TTYVTDisallocate
-ExecStart=-/sbin/agetty --noclear %I 38400
+ExecStart=-/sbin/agetty --noclear %I 38400 linux
 Type=idle
 Restart=always
 RestartSec=0
diff --git a/units/serial-getty@.service.m4 b/units/serial-getty@.service.m4
index d6a7669..2dc210e 100644
--- a/units/serial-getty@.service.m4
+++ b/units/serial-getty@.service.m4
@@ -36,8 +36,7 @@ Before=getty.target
 IgnoreOnIsolate=yes
 
 [Service]
-Environment=TERM=vt102
-ExecStart=-/sbin/agetty -s %I 115200,38400,9600
+ExecStart=-/sbin/agetty -s %I 115200,38400,9600 vt102
 Type=idle
 Restart=always
 RestartSec=0
